File: FileHandlingExceptions/Classes/FileOperations.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using BaseConnectionLibrary;

namespace FileHandlingExceptions.Classes
{
    public class FileOperations : BaseExceptionProperties
    {
        public List<string> ReadFileWithBaseExceptionProperties(string fileName)
        {
            var lines = new List<string>();

            try
            {
                lines = File.ReadAllLines(fileName).ToList();
            }
            catch (Exception e)
            {
                mHasException = true;
                mLastException = e;
            }
            return lines;
        }
        public List<string> ReadFileWithOutBaseExceptionProperties(string fileName) => 
            File.ReadAllLines(fileName).ToList();
    }
}

File: WindowsApp1/Classes/FileOperations.vb
Imports System.IO
Imports BaseConnectionLibrary

Public Class FileOperations
    Inherits BaseExceptionProperties
    Public Function ReadFileWithBaseExceptionProperties(ByVal fileName As String) As List(Of String)
        mHasException = False
        Dim lines = New List(Of String)()

        Try
            lines = File.ReadAllLines(fileName).ToList()
        Catch e As Exception
            mHasException = True
            mLastException = e
        End Try
        Return lines
    End Function

End Class

